torch/_inductor/kernel/mm_common.py

```diff
@@ -284,17 +295,19 @@ def mm_args(mat1, mat2, *others, layout=None, out_dtype=None, use_4x2_dim=False)
 
         if out_dtype is None:
             out_dtype = mat1.get_dtype()
+
         layout = FixedLayout(
             mat1.get_device(),
             out_dtype,
             [*b, m, n],
         )
     else:
         assert out_dtype is None, "out_dtype is ignored if layout is specified."
-
     from ..lowering import expand
 
-    others = [realize_inputs(expand(x, layout.size)) for x in others]
+    # For WoQ GEMM, we can't expand scale because the corresponding ATen op doesn't support 2d scale.
+    if not (mat1.get_dtype() == torch.bfloat16 and mat2.get_dtype() == torch.int8):
```

Comment:
Only BF16 activations have been supported in this PR because for FP16 & FP32, weight is dequantized during constant-folding pass of freezing, and then after auto-tuning, performance with a large `M` dimension may be better than either `torch.ops.aten._weight_int8pack_mm`, or the WoQ micro-kernel support introduced in this PR, which dequantizes `w` within the micro-kernel.

While even BF16 activations with a large `M` dimension may benefit from dequantizing `w` beforehand, for now, they would use WoQ support in GEMM templates for auto-tuning, and then a subsequent PR would add logic for deciding whether or not to dequantize weights beforehand.